sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,648 @@
+typedef enum dif_keymgr_toggle {
+  /**
+   * Enabled state.
+   */
+  kDifKeymgrToggleEnable,
+  /**
+   * Disabled state.
+   */
+  kDifKeymgrToggleDisable,
+} dif_keymgr_toggle_t;
+
+/**
+ * Hardware instantiation parameters for key manager.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_keymgr_params {
+  /**
+   * Base address of key manager registers.
+   */
+  mmio_region_t base_addr;
+} dif_keymgr_params_t;
+
+/**
+ * A handle to key manager.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_keymgr {
+  /**
+   * Hardware instantiation parameters.
+   */
+  dif_keymgr_params_t params;
+} dif_keymgr_t;
+
+/**
+ * Result of a key manager operation.
+ */
+typedef enum dif_keymgr_result {
+  /**
+   * The call succeeded.
+   */
+  kDifKeymgrOK = 0,
+  /**
+   * A non-specific error occurred and the hardware is in an invalid or
+   * irrecoverable state.
+   */
+  kDifKeymgrError = 1,
+  /**
+   * The caller supplied invalid arguments but the call did not cause any
+   * side-effects and the hardware is in a valid and recoverable state.
+   */
+  kDifKeymgrBadArg = 2,
+} dif_keymgr_result_t;
+
+/**
+ * Result of a key manager operation that writes to lockable registers.
+ */
+typedef enum dif_keymgr_lockable_result {
+  /**
+   * The call succeeded.
+   */
+  kDifKeymgrLockableOK = kDifKeymgrOK,
+  /**
+   * A non-specific error occurred and the hardware is in an invalid or
+   * irrecoverable state.
+   */
+  kDifKeymgrLockableError = kDifKeymgrError,
+  /**
+   * The caller supplied invalid arguments but the call did not cause any
+   * side-effects and the hardware is in a valid and recoverable state.
+   */
+  kDifKeymgrLockableBadArg = kDifKeymgrBadArg,
+  /**
+   * The register that needs to be written to is locked.
+   */
+  kDifKeyMgrLockableLocked,
+} dif_keymgr_lockable_result_t;
+
+/**
+ * Key manager interrupts.
+ */
+typedef enum dif_keymgr_irq {
+  /**
+   * Operation was completed.
+   *
+   * Note: This interrupt is triggered regardless of the outcome of the
+   * operation. An operation is successful if only `kDifKeymgrIrqDone` is
+   * triggered and unsuccessful if both `kDifKeymgrIrqDone` and
+   * `kDifKeymgrIrqError` are triggered.
+   */
+  kDifKeymgrIrqDone,
+  /**
+   * Key manager encountered an error while performing the operation.
+   *
+   * There are two categories of errors: hardware fault errors and software
+   * operation errors. Hardware fault errors are caused by invalid KMAC
+   * commands, states, or outputs and are not related to software. Software
+   * operation errors are caused by invalid inputs from software. Clients can
+   * use `dif_keymgr_get_status()` to determine the type of error that occurred.
+   */
+  kDifKeymgrIrqError,
+  /**
+   * \internal Last key manager interrupt.
+   */
+  kDifPwrmgrIrqLast = kDifKeymgrError,
```

Comment:
Pwrmgr? =P

I would drop this and add it only if you wind up needing it in the impl.